addons/sms/models/sms_sms.py

```diff
@@ -58,6 +58,34 @@ def send(self, delete_all=False, auto_commit=False, raise_exception=False):
             if auto_commit is True and not getattr(threading.currentThread(), 'testing', False):
                 self._cr.commit()
 
+    def retry(self):
+        self.state = 'outgoing'
+
+    def resend(self):
+        sms_to_send = self.filtered(lambda sms: sms.state == 'error')
+        notification_title = _('Warning')
+        notification_type = 'danger'
+        notification_message = ''
+        if sms_to_send:
+            sms_to_send.send()
+            if len(sms_to_send) > len(sms_to_send.exists()):
+                notification_title = _('Success')
+                notification_type = 'success'
+                notification_message = _('The selected SMS Text Messages are successfully being resent.')
```

Comment:
If at least one sms is send, ok to display a green toaster. However if some are failed, could be great to tell "4 SMS are resent and 2 failed", otherwise people may think everything went well but actually only 1 of their 200 sms was sent.

Reply:
Dev Done !